library/alloc/src/vec/mod.rs

```diff
@@ -2695,7 +2666,7 @@ impl<T: Clone, A: Allocator + Clone> Clone for Vec<T, A> {
     }
 
     fn clone_from(&mut self, other: &Self) {
-        SpecCloneFrom::clone_from(self, other)
+        crate::slice::SpecCloneIntoVec::clone_into(other.as_slice(), self);
```

Comment:
suggestion: How about this becomes
```suggestion
        other.as_slice().clone_into(self);
```
as that'll call the specialization trait anyway...

Reply:
Thanks for the feedback!

Sadly this doesn’t compile. The problem is that the `Owned` type for the `ToOwned` implementation on a slice is `Vec<T>` which is the same as `Vec<T, Global>`, whereas in the `vec::clone_from` implementation `Self` is `Vec<T, A>`, ie, it needs to work with any allocator, so the types don’t match when we try to call `clone_into(self)`. That is probably why the person who made 8725e4c337 decided to duplicate the code to work around the problem.

Reply:
Oh, good observation!  As you have it makes sense then.

(I also started a thread in <https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/Allocators.20on.20.60to_owned.60.3F/near/324517903> about whether `to_owned` should support allocators somehow.)